flexget/plugins/input/kitsu.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.cached_input import cached
+from requests import RequestException
+
+log = logging.getLogger('kitsu')
+
+
+class KitsuAnime(object):
+    """Creates an entry for each item in your kitsu.io list.
+    Syntax:
+    kitsu:
+      username: <value>
+      lists:
+      	- <current|planned|completed|on_hold|dropped>
+      	- <current|planned|completed|on_hold|dropped>
+      latest: <yes|no>
+      finishedonly: <yes|no>
+      currentonly: <yes|no>
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'username': {'type': 'string'},
+            'lists': {'type': 'array',
+                      'items': {'type': 'string', 'enum': ['current', 'planned', 'completed', 'on_hold', 'dropped']}},
+            'latest': {'type': 'boolean', 'default': False},
+            'currentonly': {'type': 'boolean', 'default': False},
+            'finishedonly': {'type': 'boolean', 'default': False}
+        },
+        'required': ['username'],
+        'additionalProperties': False,
+    }
+
+    @cached('kitsu', persist='2 hours')
+    def on_task_input(self, task, config):
+        entries = []
+        try:
+            user_payload = {'filter[name]': config['username']}
+            user_response = task.requests.get('https://kitsu.io/api/edge/users', params=user_payload)
+            user_response.raise_for_status()
+            user = user_response.json()
+            if len(user['data']) < 1:
+                raise plugin.PluginError('no such username found on kitsu.io')
+            userId = user['data'][0]['id']
+        except RequestException:
+            raise plugin.PluginError('Error getting User ID from kitsu.io')
+
+        next_url = 'https://kitsu.io/api/edge/users/{userId}/library-entries'.format(**locals())
```

Comment:
I'd prefer not to use `locals()` even though it's fairly explicit what's happening in this case.